### ISSUE TITLE
I'm back! (What I learned in the meantime) - NOT A ISSUE,  just a greeting and a welcome back to you all :)

### ISSUE THREAD

**Issue by RobertoMilani** [rungwiroon/BlazorGoogleMaps#68]
Sorry for the long absence: some summer vacation and some SW releases at work.
In my spare time, I devoted myself to studying Blazor Web Assembly, which until today I had "snubbed".

Well, let me tell you what a great platform it is!
Obviously I'm still learning, but the mere fact of having a cross-platform client, focused on presentation, which communicates with REST API on the server which in turn can call a Service class, allows you to have your own imagination as the only limit to what you can implement!

For heaven's sake, Blazor server applications are not bad, but especially from a mobile point of view, they suffer heavily from any connection losses.

Moreover, in my case, having them hosted on Raspebrry Pi and behind a NGINX proxy it happens that for no apparent reason, from time to time the WebSocket stops, leaving the application in freeze for a few seconds.

It seems that this is due to the sum of the following conditions:
a) Hosting on Raspberry pi
b) NGINX Proxy
c) Use of HTTPS
Needless to say, I also tested on a notebook with Windows 10 and with IIS Server and the problem nevertheless occurred. The only factor in common, the HTTPS which I certainly do not want to give up.

So I decided to switch padadigma and switch to Blazor WebAssembly!

And from what I have been able to read around, my hopes seem well pinned!

---

And now to us ...
What good is all this introduction in our context?

**Well, all this just for one survey:
how many of you have already used the library in WebAssembly projects and how many in Blazor Server projects?**

At the moment I'm trying (it's a huge job) to create a Spotify client in Blazor WebAssebly.
If I'm lucky and good and a good project comes out, who knows I'll put it on GitHub as well :D

In the meantime "happy coding" to all of you and good use of BlazorGoogleMaps !!! :)

**Comment by valentasm1** [comment]
Happy to hear that you had holidays. Of course like real dev doing some programming while on it. I do the same 👍 

Personally i only have couple of websites running on Server Side Blazor. I just didnt liked development experience with Blazor WebAssembly. Of course it was long time ago and i sure it changed to better side.

**Comment by turner11** [comment]
At work I'm developing a server side app and I have also a personal WebAssembly site.

As @valentasm1  said, the experience of developing a Blazor WebAssembly app was sub optimal, but there is a great work around for having the best of both worlds.

[This](https://www.youtube.com/watch?v=SkYQDPXw__c&list=PL8h4jt35t1wjvwFnvcB2LlYL4jLRzRmoz&index=7&t=0s&ab_channel=DevExpress) video (sorry, could not find a written tutorial) explains how you can develop a single app and run it using server / client side.    
The tutorial will take you 90% of the way, I had to do some modifications for keeping the content of `wwwroot` in sync in both projects (I used build events).  

Using this methodology, I was able to build the app using the great debug tooling that supports server side, and then deploy the WebAssembly app as a static site, thus pushing the costs down to virtually 0.

**Comment by valentasm1** [comment]
@turner11 Exacly. I knew that changing from ServerSide > Web assembly is very minimal. Thank you for the video

**Comment by RobertoMilani** [comment]
Thanks for your replies! :D
I "worked hard" during last days on Blazor WebAssemby core hosted solutions and...

let me tell you that the more I know him the more I like him!
I solved all notification persistence issues and was able to create the first draft of my Spotify music player! : D

At the moment I am "arguing" with the renewal of the token, which in certain situations does not work, but even just seeing playlists and "current playing" synchronized with my Spotify account I consider it a success!

Not only that ... I also learned how to convert a Blazor Server application into a WebAssembly, with the only limitation, in my specific case, that I have to have a .Net Core Hosted part to allow it to be published on Raspberry (the WebAssembly part natively would be .Net Standard 2.x, therefore not releasable on Raspberry, while the .Net Core Hosted part allows you to encapsulate everything inside a .Net Core executable natively executable on Raspberry).

And - demonstrating how light it is on the server side - here are the Glances statistics of the two parts of the application, the authentication service and the App itself: CPU between 0 and 0.2% and very little RAM, despite the client player is running.

---


Well, sorry, I have dwelt ... because of my enthusiasm!
And this is no place to discuss the various Blazor implementations, however, after sharing the survey with you, it felt right to include my comments :)

See you soon!
